tools/transformer/src/transform.js

```diff
@@ -13,6 +14,7 @@ log.setLevel(process.env.XGEN_LOGGING_LEVEL || "warn");
 
 let { doc, apiFileLocation } = getAPI(process.argv.slice(2));
 
+doc = applyDiscriminatorTransformations(doc);
```

Comment:
Order of that operation is important. We need to run this before oneOf and allOf transformation as those will be picking up oneOfs introduced by discriminatorTransformation :)